File: homework-5/code/oddGenerator.cpp
#include <iostream>
using namespace std;

class OddGenerator {
	private:
		int x = -1;
	public:
		int nextOdd();
};

int OddGenerator::nextOdd() {
	return x += 2;
}

int main () {
	OddGenerator odds;
	cout << odds.nextOdd() << endl;
	cout << odds.nextOdd() << endl;
	cout << odds.nextOdd() << endl;
	cout << odds.nextOdd() << endl;
}

File: homework-5/code/OddGenerator.java
public class OddGenerator {
	
	private int x = -1;
	public int nextOdd() {
		return x += 2;
	}

	public static void main (String[] args) {
		OddGenerator odds = new OddGenerator();
		System.out.println(odds.nextOdd());
		System.out.println(odds.nextOdd());
		System.out.println(odds.nextOdd());
		System.out.println(odds.nextOdd());
	}
}

